src/object_storage/protocol.rs

```diff
@@ -81,10 +83,12 @@ impl Object {
                 )
             })?;
         let ct = protocol::get_header(value, &CONTENT_TYPE)?.map(From::from);
+        let hash = protocol::get_header(value, &ETAG)?.unwrap();
```

Comment:
Please don't use `unwrap` unless you're 100% sure it's not possible to get None. I think in this case it is, the documentation says "For objects smaller than 5 GB, this value is the MD5 checksum of the object content. ". It's not clear what happens in an object is larger than 5 GB. I think we need to expose `Option<String>` for the hash.

Reply:
Objects larger than 5GB are stored in "manifests objects", and they also have an etag/hash, but calculated in another way:

> For manifest objects, this value is the MD5 checksum of the concatenated string of ETag values for each of the segments in the manifest, and not the MD5 checksum of the content that was downloaded.

But I change the type into `Option<String>`, which should be more secure.